force-app/main/default/classes/ClassesObjectsMethods.cls

```diff
@@ -212,10 +244,11 @@ public with sharing class ClassesObjectsMethods {
      * @return The a positive number years between the two dates, or null if either date is null.
      */
     public static Integer calculateYearDifference(Date d1, Date d2) {
-
+        if (d1 == null || d2 == null) {
+            return null;
+        }
         // Subtract the difference in years
-
-        return null; // Replace null with the variable you used to store the result
+        return (d2.year() - d1.year());
```

Comment:
good job. The problem was not totally clear here. So in my solution is wrapped it in a call to Math.abs. But this passes the tests!